activesupport/lib/active_support/core_ext/string/output_safety.rb

```diff
@@ -136,15 +136,39 @@ def encode_with(coder)
 
     UNSAFE_STRING_METHODS.each do |unsafe_method|
       if unsafe_method.respond_to?(unsafe_method)
+        arity = unsafe_method.method(unsafe_method).arity
+        params =
+          case arity
+          when 0
+            "&block"
+          when 1
+            "arg, &block"
+          when (2..)
+            1.upto(arity).each_with_object(+"") { |i, s| s << "arg#{i}, " } << "&block"
+          else
+            "*args, &block"
+          end
+        params_for_super =
+          case arity
+          when 0
+            nil
+          when 1
+            "_"
+          when (2..)
+            1.upto(arity).map { |i| "_arg#{i}" }.join(", ")
+          else
+            "*"
+          end
+
         class_eval <<-EOT, __FILE__, __LINE__ + 1
-          def #{unsafe_method}(*args, &block)       # def capitalize(*args, &block)
-            to_str.#{unsafe_method}(*args, &block)  #   to_str.capitalize(*args, &block)
-          end                                       # end
-
-          def #{unsafe_method}!(*args)              # def capitalize!(*args)
-            @html_safe = false                      #   @html_safe = false
-            super                                   #   super
-          end                                       # end
+          def #{unsafe_method}(#{params})       # def capitalize(*args, &block)
+            to_str.#{unsafe_method}(#{params})  #   to_str.capitalize(*args, &block)
+          end                                   # end
+
+          def #{unsafe_method}!(#{params_for_super})     # def capitalize!(*args)
```

Comment:
Is there a benefit to having different `params_for_super` here?  Could we just reuse `params`?

```suggestion
          def #{unsafe_method}!(#{params})      # def capitalize!(*args)
```

Reply:
I prefer not to give unused parameter a name. There may be rooms for optimization in the VM for such pattern in the future. BTW the comment on the right side still displays `*args`, but that was my mistake. Just updated the comment.